apm-agent-plugins/apm-jms-plugin/src/test/java/co/elastic/apm/agent/jms/Jms1InstrumentationIT.java

```diff
@@ -0,0 +1,127 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.jms;
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.activemq.pool.PooledConnectionFactory;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageListener;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.Topic;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * ActiveMQ tests for JMS 1 API.
+ * Testing with the Pooled connection factory adds testing for name-based pre-matcher-filter, and tests for nested receives
+ */
+public class Jms1InstrumentationIT extends AbstractJmsInstrumentationIT {
+
+    private static Connection connection;
+    private Session session;
+
+    @BeforeClass
+    public static void startQueueAndClient() throws JMSException {
+        ConnectionFactory connectionFactory = new PooledConnectionFactory(
+            new ActiveMQConnectionFactory("vm://localhost?broker.persistent=false"));
+        connection = connectionFactory.createConnection();
+        connection.start();
+    }
+
+    @AfterClass
+    public static void stopQueueAndClient() throws JMSException {
+        connection.stop();
+    }
+
+    @Before
+    public void startSession() throws JMSException {
+        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+    }
+
+    @After
+    public void closeSession() throws JMSException {
+        // This should also close underlying producers and consumers
+        session.close();
+    }
+
+    @Override
+    protected Queue createQueue(String queueName) throws JMSException {
+        return session.createQueue(queueName);
+    }
+
+    @Override
+    protected Topic createTopic(String topicName) throws JMSException {
+        return session.createTopic(topicName);
+    }
+
+    @Override
+    protected Message createTextMessage(String messageText) throws JMSException {
+        return session.createTextMessage(messageText);
+    }
+
+    @Override
+    protected void send(Destination destination, Message message) throws JMSException {
+        session.createProducer(destination).send(message);
+    }
+
+    @Override
+    protected CompletableFuture<Message> registerListener(Destination destination) throws JMSException {
+        MessageConsumer consumer = session.createConsumer(destination);
+        final CompletableFuture<Message> incomingMessageFuture = new CompletableFuture<>();
+        //noinspection Convert2Lambda,Anonymous2MethodRef - we don't instrument lamdas or method references
```

Comment:
We should probably wrap lambdas so that we can instrument the wrapper. Otherwise, we'd miss out on those.

Reply:
Good idea. I already have a helper in place for pooling wrappers 😉 
As far as I can see, APIs I will need to instrument for that are:
1. `MessageConsumer#setMessageListener` (following the same logic of supporting JMS 2 APIs through JMS 1.1 APIs, this should cover `JMSConsumer#setMessageListener` as well)
2. `Session#setMessageListener`

Will need to see further if one covers the other, so maybe I can get away with only one.

Reply:
Looks like `Session#setMessageListener` is not an API to be used in plain JMS clients, more something for usage by app servers or the like. So, currently I won't instrument it.